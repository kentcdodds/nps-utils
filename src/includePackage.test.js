import {includePackage, commonTags} from './index'

test('given just a string, it looks for scripts at a default location', () => {

  jest.mock(`./packages/foo/package-scripts.js`, () => ({
    scripts: {
      default: 'echo foo',
    },
  }), {virtual: true})

  jest.mock(`./packages/bar/package-scripts.js`, () => ({
    scripts: {
      default: 'echo bar',
    },
  }), {virtual: true})

  expect(includePackage('foo'))
    .toEqual({default: 'cd packages/foo && npm start default'})

  expect(includePackage('bar'))
    .toEqual({default: 'cd packages/bar && npm start default'})

})

test(commonTags.oneLine`given just a string, if that package doesnt exist,
  it thows a helpful error`, () => {
  
  expect(() => includePackage('doesnt-exist'))
    .toThrowError(/Couldnt include the package/)
  
})

test('given an explicit path, it uses that', () => {
  
  jest.mock(`./hello/explicit-foo.js`, () => ({
    scripts: {
      default: 'echo foo',
    },
  }), {virtual: true})

  jest.mock(`./explicit-bar.js`, () => ({
    scripts: {
      default: 'echo bar',
    },
  }), {virtual: true})

  expect(includePackage({path: `./hello/explicit-foo.js`}))
    .toEqual({default: 'cd hello && npm start default'})

  expect(includePackage({path: `./explicit-bar.js`}))
    .toEqual({default: 'cd  && npm start default'})

})

test(commonTags.oneLine`given an explicit path, if that package doesnt exist,
  it thows a helpful error`, () => {

  expect(() => includePackage({path: 'it-doesnt-exist.js'}))
    .toThrowError(/Couldnt include the package/)

})

test('given complex package, correct includes all scripts', () => {
  
  jest.mock(`./scripts/complex-package.js`, () => ({
    scripts: {
      default: 'nps foo',
      foo: 'echo foo',
      bar: 'echo foo',
      nested: {
        default: 'nps nested.bbb',
        aaa: 'echo aaa',
        bbb: 'echo bbb',
      },
    },
  }), {virtual: true})

  expect(includePackage({path: `./scripts/complex-package.js`}))
    .toMatchSnapshot()

})

test('description nodes respected', () => {
  
  jest.mock(`./scripts/complex-package-description.js`, () => ({
    scripts: {
      foo: 'echo foo',
      nested: {
        script: 'echo hia',
        description: 'this is nested',
      },
    },
  }), {virtual: true})

  
  expect(includePackage({path: './scripts/' +
    'complex-package-description.js'}))
    .toMatchSnapshot()

})

