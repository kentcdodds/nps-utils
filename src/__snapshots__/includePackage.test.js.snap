// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description nodes respected 1`] = `
Object {
  "foo": "cd scripts\\\\subdir && npm start foo",
  "nested": Object {
    "description": "this is nested",
    "script": "cd scripts\\\\subdir && npm start nested && cd \\"..\\\\..\\"",
  },
}
`;

exports[`given complex package, correct includes all scripts 1`] = `
Object {
  "bar": "cd scripts && npm start bar",
  "default": "cd scripts && npm start default",
  "foo": "cd scripts && npm start foo",
  "nested": Object {
    "aaa": "cd scripts && npm start nested.aaa",
    "bbb": "cd scripts && npm start nested.bbb",
    "default": "cd scripts && npm start nested.default",
  },
}
`;
